### PR TITLE
MINOR: Handle some null cases in BrokerMetadataPublisher

### PR DESCRIPTION
**Pull request by mumrah**
Found a few NPEs when running system tests. Added coverage for a few null cases in BrokerMetadataPublisher as well as a unit test

### CODE REVIEW

**Review by cmccabe — APPROVED** [review]
LGTM. Thanks, @mumrah .

File: core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala [hunk 1]
```diff
@@ -46,8 +46,10 @@ object BrokerMetadataPublisher {
   def getTopicDelta(topicName: String,
                     newImage: MetadataImage,
                     delta: MetadataDelta): Option[TopicDelta] = {
-    Option(newImage.topics().getTopic(topicName)).map {
-      topicImage => delta.topicsDelta().changedTopic(topicImage.id())
+    Option(newImage.topics().getTopic(topicName)).flatMap {
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala:49]: Is `newImage.topics()` never null? If so, should we add null check to constructor of `MetadataImage` to make sure input arguments are NOT null.

**Review by chia7712 — APPROVED** [review]
LGTM. one trivial comment below

File: core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala (hunk 1 above)
Comment (mumrah) [reply core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala:49]: There's just two usages of this constructor and they cannot provide nulls, so we can probably live without the null check

**Comment by mumrah** [comment]
Cherry-picked to 3.0 as 89c9c94f877

**Pull request merged**